File: back/src/main/java/com/pingo/entity/users/UserInfo.java
package com.pingo.entity.users;

import com.pingo.exception.BusinessException;
import com.pingo.exception.ExceptionCode;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.time.LocalDateTime;
import java.util.regex.Pattern;

@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class UserInfo {
    private String userNo;
    private LocalDateTime userBirth;
    private int userHeight;
    private String userAddress;
    private String user1stJob;
    private String user2ndJob;
    private String userReligion;
    private String userDrinking;
    private String userSmoking;
    private String userBloodType;

    public UserInfo validatedSignUpUserInfoData() {

        // 검증하는 로직
        // 1. 생년월일 검증
        // 생년월일은 오늘 날짜보다 이후일 수 없음
        if (this.userBirth.isAfter(LocalDateTime.now())) {
            throw new BusinessException(ExceptionCode.INVALID_USER_BIRTH);
        }

        return this;
    }

    public void insertUserNo(String userNo) {
        this.userNo = userNo;
    }
}

File: front/lib/data/models/user_model/user_info.dart
class UserInfo {
  String? userNo;
  DateTime? userBirth;
  int? userHeight;
  String? userAddress;
  String? user1stJob;
  String? user2ndJob;
  String? userReligion;
  String? userDrinking;
  String? userSmoking;
  String? userBloodType;

  UserInfo({
    this.userNo,
    this.userBirth,
    this.userHeight,
    this.userAddress,
    this.user1stJob,
    this.user2ndJob,
    this.userReligion,
    this.userDrinking,
    this.userSmoking,
    this.userBloodType,
  });

  Map<String, dynamic> toJson() {
    return {
      "userNo": userNo,
      "userBirth": userBirth?.toIso8601String(),
      "userHeight": userHeight,
      "userAddress": userAddress,
      "user1stJob": user1stJob,
      "user2ndJob": user2ndJob,
      "userReligion": userReligion,
      "userDrinking": userDrinking,
      "userSmoking": userSmoking,
      "userBloodType": userBloodType,
    };
  }

  UserInfo.fromJson(Map<String, dynamic> json)
      : userNo = json['userNo'],
        userBirth = json['userBirth'] != null
            ? DateTime.parse(json['userBirth'])
            : null,
        userHeight = json['userHeight'],
        userAddress = json['userAddress'],
        user1stJob = json['user1stJob'],
        user2ndJob = json['user2ndJob'],
        userReligion = json['userReligion'],
        userDrinking = json['userDrinking'],
        userSmoking = json['userSmoking'],
        userBloodType = json['userBloodType'];

  @override
  String toString() {
    return 'UserInfo{userNo: $userNo, userBirth: $userBirth, userHeight: $userHeight, userAddress: $userAddress, user1stJob: $user1stJob, user2ndJob: $user2ndJob, userReligion: $userReligion, userDrinking: $userDrinking, userSmoking: $userSmoking, userBloodType: $userBloodType}';
  }

  // .copywith 메서드 생성( 깊은 복사로 복사본 만드는 함수 )
  UserInfo copyWith(UserInfo? oldInfo) {
    return UserInfo(
      userNo: oldInfo?.userNo ?? this.userNo,
      userBirth: oldInfo?.userBirth ?? this.userBirth,
      userHeight: oldInfo?.userHeight ?? this.userHeight,
      userAddress: oldInfo?.userAddress ?? this.userAddress,
      user1stJob: oldInfo?.user1stJob ?? this.user1stJob,
      user2ndJob: oldInfo?.user2ndJob ?? this.user2ndJob,
      userReligion: oldInfo?.userReligion ?? this.userReligion,
      userDrinking: oldInfo?.userDrinking ?? this.userDrinking,
      userSmoking: oldInfo?.userSmoking ?? this.userSmoking,
      userBloodType: oldInfo?.userBloodType ?? this.userBloodType,
    );
  }
}
